iml-agent/src/rpm.rs

```diff
@@ -1,25 +1,51 @@
 use regex::Regex;
 
+use std::fmt;
 use std::process::Output;
 
 use crate::{agent_error::ImlAgentError, cmd::cmd_output};
 
-fn parse(output: Output) -> Result<bool, ImlAgentError> {
+#[derive(Debug, Eq, PartialEq, serde::Serialize, serde::Deserialize)]
+pub struct Version(String);
+
+impl fmt::Display for Version {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+async fn parse(output: Output) -> Result<Option<Version>, ImlAgentError> {
     if output.status.success() {
-        Ok(true)
+        // In case there's syntax error in query format, exit code of `rpm` is 0,
+        // but there's no data and an error is on stderr
+        if output.stderr.len() > 0 {
+            Err(ImlAgentError::CmdOutputError(output))
+        } else {
+            Ok(Some(Version(
+                String::from_utf8_lossy(&output.stdout).to_string(),
+            )))
+        }
     } else {
         let stdout = output.stdout.clone();
         let re = Regex::new(r"^package .*? is not installed\n$").unwrap();
         let s = String::from_utf8(stdout)?;
         if re.is_match(&s) {
```

Comment:
I know this is not part of this patch, but since you're changing things in here.. we could avoid cloning stdout by doing this:

```rust
        let re = Regex::new(r"^package .*? is not installed\n$").unwrap();
        let s = std::str::from_utf8(&output.stdout)?;
        if re.is_match(s) {
```

Reply:
Nice suggestion, done